tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EnumSuite.scala

```diff
@@ -385,6 +385,105 @@ class EnumSuite extends BaseDottySuite {
     )
   }
 
+  test("enum-self-braces") {
+    val code =
+      """|enum A { self =>
+         |  case B(v: Int)
+         |  case C(v: String)
+         |}
+      """.stripMargin
+    runTestAssert[Stat](code)(
+      Defn.Enum(
+        Nil,
+        Type.Name("A"),
+        Nil,
+        Ctor.Primary(Nil, Name(""), Nil),
+        Template(
+          Nil,
+          Nil,
+          Self(Term.Name("self"), None),
+          List(
+            Defn.EnumCase(
+              Nil,
+              Term.Name("B"),
+              Nil,
+              Ctor.Primary(
+                Nil,
+                Name(""),
+                List(List(Term.Param(Nil, Term.Name("v"), Some(Type.Name("Int")), None)))
+              ),
+              Nil
+            ),
+            Defn.EnumCase(
+              Nil,
+              Term.Name("C"),
+              Nil,
+              Ctor.Primary(
+                Nil,
+                Name(""),
+                List(List(Term.Param(Nil, Term.Name("v"), Some(Type.Name("String")), None)))
+              ),
+              Nil
+            )
+          ),
+          Nil
+        )
+      )
+    )
+  }
+
+  test("enum-self-indented") {
+    val code =
+      """|enum A:
```

Comment:
This is basically the same as:
```scala
enum A:
   self =>
   case B(v: Int)
   case C(v: String)
```

Indentation after `self =>` doesn't seem to be playing any role. We should most likely add a test case there too.